models/credmark/algorithms/var_engine.py

```diff
@@ -0,0 +1,289 @@
+from typing import (
+    List,
+    Optional,
+    Dict,
+    Union,
+)
+
+from datetime import (
+    datetime,
+    timezone,
+    date,
+)
+
+import credmark.model
+
+from credmark.model import ModelRunError
+
+from credmark.dto import (
+    DTO,
+    DTOField,
+)
+
+from credmark.types import (
+    Portfolio,
+    Price,
+    Address,
+)
+
+from models.credmark.algorithms.risk import (
+    PortfolioManager,
+    TokenTradeable,
+    Market,
+    calc_var,
+)
+
+import numpy as np
+import pandas as pd
+
+
+class PriceList(DTO):
+    price: Price
+    token: Address
+
+
+class ContractVaRInput(DTO):
```

Comment:
what is the contract part of this?

Reply:
If calculating VaR for a contract, the position is supplied by the debt/asset levels in the contract. We are left with only VaR parameters to be specified. This can be renamed as "VaRParameter".